### DISCUSSION TITLE
Can write webp but cannot read webp (fails successfully by just not writing the image)

### DISCUSSION THREAD

**Discussion by cyberpwnn** [ImageMagick/ImageMagick#7706] [Help]
### ImageMagick version

7.1.1-39

### Operating system, version and so on

Docker (debian:bullseye-slim)

### Description

```Dockerfile
# Stage 1: Build ImageMagick
FROM debian:bullseye-slim AS imagemagick-builder

# Install build dependencies
RUN apt-get update && apt-get install -y \
    build-essential \
    wget \
    libltdl-dev \
    libpng-dev \
    libjpeg-dev \
    libtiff-dev \
    libwebp-dev \
    libjbig-dev \
    libgomp1 \
    pkg-config \
    && apt-get clean \
    && rm -rf /var/lib/apt/lists/*

# Download and compile ImageMagick
WORKDIR /tmp
RUN wget https://github.com/ImageMagick/ImageMagick/archive/refs/tags/7.1.1-39.tar.gz \
    && tar xvzf 7.1.1-39.tar.gz \
    && cd ImageMagick-7.1.1-39 \
    && ./configure --disable-hdri --with-quantum-depth=8 --with-png=yes --with-webp=yes --with-jpeg=yes \
    --with-jp2=yes --without-tiff --with-modules --enable-shared \
    && make \
    && make install
```

When running this through ffi, simply reading an image then writing it will succeed unless the image being read is webp, then it fails, but jpg to webp works just fine. It fails by just not writing a file, exits fine.

I have tried without shared and without modules, but it cant find libwebp.so.6, I even tried building the webp targz from scratch off the magick delegates repo and still nothing. 

Any help would be appreciated I've spent more time trying to use imagemagick than i'd like to admit, I've tried lots of variants of dependencies, configuration options, one-click installers, they all either are missing some dependency or just fail in the most undebuggable ways. 

```
...
-------------------------------------------------------------
checking for libwebp >= 0.4.1... yes
checking for libwebpmux >= 0.5.0 libwebpdemux >= 0.5.0... yes
...
```

**Comment by fmw42** [comment]
Did you get and install the correct version of libwebp?   See https://developers.google.com/speed/webp/download and https://storage.googleapis.com/downloads.webmproject.org/releases/webp/index.html

**Reply by cyberpwnn** [reply]
I have no idea what version I have, though I would assume that its at least 0.4.1 which imagemagick is checking for so I assumed it should work fine. Do I need to compile specific versions of every delegate just to make sure it functions how its supposed to? Does everyone have to do this?

**Comment by snibgo** [comment]
I don't use webp much, but have no problems reading or writing.

What is the text output from:
```
magick -list format | grep -i webp
```

Expect something like:
```
     WEBP* rw+   WebP Image Format (libwebp 1.4.0 [020F])
```
"rw+" means my build of IM can read and write webp, including multiple images within one file.

```
./configure --disable-hdri --with-quantum-depth=8 --with-png=yes --with-webp=yes ...
```
Towards the bottom of `config.log`, there is a section starting:
```
ImageMagick 7.1.1-39 is configured as follows. Verify that this configuration
matches your expectations.
```
What does that say about webp? Expect a line like this:
```
  WEBP              --with-webp=yes			yes
```

**Reply by cyberpwnn** [reply]
Best i can see from the configure log is the following lines

```
checking for libwebp >= 0.4.1...yes
checking for libwebpmux >= 0.5.0 libwebpdemux >= 0.5.0...yes
......
checking for dwebp...dwebp
checking for cwebp...cwebp
.......
WEBP              --with-webp=yes			yes
......
WEBP --with-webp=yes yes
.....
DELEGATES = jbig jng jpeg lzma png webp zlib
```

I dont see anything like      `WEBP* rw+   WebP Image Format (libwebp 1.4.0 [020F])`

**Comment by cyberpwnn** [comment]
Ok update, tried to run it through cli and got this error

```
magick: delegate failed `'dwebp' -pam '%i' -o '%o'' @ error/delegate.c/InvokeDelegate/1913.
2024-10-19T19:28:04.499704344Z magick: unable to load module '/usr/local/lib/ImageMagick-7.1.1/modules-Q8/coders/webp.la': file not found @ error/module.c/OpenModule/1293.
2024-10-19T19:28:04.499709544Z magick: unable to open file '/tmp/magick-jdd7YDCDp0H8HTnd5raythx94IE9wcqe': No such file or directory @ error/constitute.c/ReadImage/785.
```

**Comment by fmw42** [comment]
Look in your delegates.xml file and see if you have

**`<delegate decode="webp" command="&quot;dwebp&quot; -pam &quot;%i&quot; -o &quot;%o&quot;"/>`**

If so,

Then check

`magick -list format`

and report what you have for the line starting with WEBP. For example I get

**`WEBP* WEBP      rw+   WebP Image Format (libwebp 1.4.0 [020F])`**

be sure it say rw+ or rw. If no r, then your install of libwebp failed in some way or is missing something.  Reinstall and look in the config.log for the install to be sure nothing failed.

Then check

`magick list configure`

and report what it says at the line starting with Delegates.  For example I get

**DELEGATES             bzlib cairo djvu fftw fontconfig freetype gslib heic jbig jng jp2 jpeg jxl lcms ltdl lzma openexr pangocairo png ps raw rsvg tiff webp x xml zlib zstd**

Does it include webp?
